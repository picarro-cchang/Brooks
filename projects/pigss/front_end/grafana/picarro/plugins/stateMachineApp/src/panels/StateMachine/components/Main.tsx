import React, { Component } from "react";
import PicarroAPI from "../api/PicarroAPI";
import deepmerge from "deepmerge";
import { notifyError } from "../utils/Notifications";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { PlanService } from "../api/PlanService";
import PlanLayout from "./Plan/PlanLayout";
import RunLayout from "./Run/RunLayout";

const REFRESH_INTERVAL = 5;
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;
export class Main extends React.Component<any, any> {
  state = {
    initialized: false,
    modal_info: {
      show: false,
      html: "",
      num_buttons: 0,
      buttons: {}
    },
    uistatus: {},
    plan: {
      max_steps: 32,
      panel_to_show: 0,
      current_step: 1,
      focus: { row: 1, column: 1 },
      last_step: 0,
      steps: {},
      num_plan_files: 0,
      plan_files: {},
      plan_filename: "",
      bank_names: {
        1: {
          name: "",
          channels: {
            1: "Port 1",
            2: "Port 2",
            3: "Port 3",
            4: "Port 4",
            5: "Port 5",
            6: "Port 6",
            7: "Port 7",
            8: "Port 8"
          }
        },
        2: {
          name: "",
          channels: {
            1: "Port 9",
            2: "Port 10",
            3: "Port 11",
            4: "Port 12",
            5: "Port 13",
            6: "Port 14",
            7: "Port 15",
            8: "Port 16"
          }
        },
        3: {
          name: "",
          channels: {
            1: "Port 17",
            2: "Port 18",
            3: "Port 19",
            4: "Port 20",
            5: "Port 21",
            6: "Port 22",
            7: "Port 23",
            8: "Port 24"
          }
        },
        4: {
          name: "",
          channels: {
            1: "Port 25",
            2: "Port 26",
            3: "Port 27",
            4: "Port 28",
            5: "Port 29",
            6: "Port 30",
            7: "Port 31",
            8: "Port 32"
          }
        }
      }
    },
    options: {
      panel_to_show: 0
    },
    isPlan: false,
    isChanged: false,
    bankAdd: {},
    isLoaded: false,
    fileNames: [],
    isPlanning: false
  };
  constructor(props) {
    super(props);
    this.getPlanFileNames = this.getPlanFileNames.bind(this);
    this.isPlanning = this.isPlanning.bind(this);
    this.getLastRunningPlan = this.getLastRunningPlan.bind(this);
  }

  ws = new WebSocket(socketURL);

  attachWSMethods = (ws: WebSocket) => {
    this.ws.onmessage = evt => {
      // on receiving a message, add it to the list of messages
      this.handleData(evt.data);
    };

    this.ws.onclose = event => {
      if (event.code !== 1000) {
        this.setupWSComm();
      }
    };
  };

  setupWSComm = () => {
    if (this.ws.CLOSED || this.ws.readyState === 0) {
      setTimeout(() => {
        toast.dismiss();
        notifyError("Web Socket Disconnected. Retrying to connect the server.");
        this.ws = new WebSocket(socketURL);
        this.attachWSMethods(this.ws);
        this.getDataViaApi();
      }, REFRESH_INTERVAL * 1000);
    }
  };

  componentDidMount() {
    this.getDataViaApi();
    this.getPlanFileNames();
    this.getLastRunningPlan();
    this.attachWSMethods(this.ws);
  }

  componentWillUnmount() {
    console.log("Component will unmount");
    if (this.ws && this.ws.readyState === 1) {
      this.ws.send("CLOSE");
      this.ws.close(1000, "Client Initited Connection Termination");
    }
  }

  getDataViaApi = () => {
    const uiStatusData = PicarroAPI.getRequest(
      `http://${apiLoc}/uistatus`
    ).then(response => {
      response.json().then(obj => {
        this.setState(deepmerge(this.state, { uistatus: obj }));
      });
    });
    Promise.all([uiStatusData]).then(() => {
      this.setState(deepmerge(this.state, { initialized: true }));
    });
  };


  handleData(data: any) {
    const o = JSON.parse(data);
    console.log("WS ", o)
    if (this.state.initialized) {
      if ("uistatus" in o) {
        const uistatus = deepmerge(this.state.uistatus, o.uistatus);
        this.setState({ uistatus });
      }
      // } else if ("modal_info" in o) {
      //   const modal_info = deepmerge(this.state.modal_info, o.modal_info);
      //   this.setState({ modal_info });
      // }
    }
  }

  ws_sender = (o: object) => {
    this.ws.send(JSON.stringify(o));
  };

  getPlanFileNames() {
    PlanService.getFileNames().then((repsonse: any) => 
    repsonse.json().then(planfiles => {
      this.setState({ fileNames: planfiles['plans'] }, () => console.log(this.state.fileNames));
    })
    )
  }

  planFileNameUpTop() {
    if (this.state.uistatus["plan_loop"] == "ACTIVE" || this.state.uistatus["plan_run"] == "ACTIVE") {
      return true
    } else {
      return false
    }
  }

  isPlanning() {
      this.setState({isPlanning: !this.state.isPlanning})
  }

  getLastRunningPlan() {
    PlanService.getLastRunning().then((response: any) => 
      response.json().then(data => {
        if (data["message"]) {
          console.log("No Plan I suppsoe")
        }          
        else {
          this.setState({plan: data['details']});
        }

      }))
  }

  render() {
  
    return (
        <div> 
            {this.state.isPlanning ? (
                <PlanLayout
                    layoutSwitch={this.isPlanning}
                    fileNames={this.state.fileNames}
                    plan={this.state.plan}
                    uistatus={this.state.uistatus}
                    ws_sender={this.ws_sender}
                    getPlanFileNames={this.getPlanFileNames}
                />
            ):(
                <RunLayout
                    layoutSwitch={this.isPlanning}
                    fileNames={this.state.fileNames}
                    plan={this.state.plan}
                    uistatus={this.state.uistatus}
                    ws_sender={this.ws_sender}
                />
            )}

      
        </div>
    );
  }
}


  //     <Modal
        //   styles={{ overlay: { color: "black" } }}
        //   open={this.state.modal_info.show}
        //   onClose={() => this.ws_sender({ element: "modal_close" })}
        //   center
        // >
        //   <div style={{ margin: "20px" }}>
        //     <div
        //       dangerouslySetInnerHTML={{ __html: this.state.modal_info.html }}
        //     ></div>
        //   </div>
        //   {modalButtons}
        // </Modal>
        // <ToastContainer />
