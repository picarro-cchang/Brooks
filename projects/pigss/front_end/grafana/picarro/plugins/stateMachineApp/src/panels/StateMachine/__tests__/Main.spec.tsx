import React from "react";
import { mount } from "enzyme";
import "jest-styled-components";
import WS from "jest-websocket-mock";
import { Main } from "../components/Main";
import PicarroAPI from "../api/PicarroAPI";
import "jest-fetch-mock";
import { PlanPanelTypes } from "./../components/types";
import "react-toastify/dist/ReactToastify.css";
import { notifyError, notifySuccess } from "../utils/Notifications";
import mockPlan from "./../api/__mocks__/mockPlan.json";
import mockPlan2 from "./../api/__mocks__/mockPlanPanel.json";
import data from "./../api/__mocks__/mockData.json";
import mockModal from "./../api/__mocks__/mockModalInfo.json";

jest.mock("./../api/PicarroAPI.ts");
jest.mock("./../api/PlanService.ts")

const defaultState = {
  initialized: false,
  modal_info: {
    show: false,
    html: "",
    num_buttons: 0,
    buttons: {}
  },
  uistatus: {},
  plan: mockPlan2,
  options: {
    panel_to_show: 0
  },
  isPlan: true,
  isChanged: false
};

const uistatus = {
  initialized: true,
  uistatus: data
};

const modal_info = {
  initialized: false,
  modal_info: mockModal
};

const plan = {
  initialized: true,
  plan: mockPlan
};

const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://localhost:1234/ws`;
const mockWS = jest.fn(element => {
  return element;
});
const server = new WS(socketURL);
const client = new WebSocket(socketURL);

describe("<Main />", () => {
  const handleDataSpy = jest.spyOn(Main.prototype, 'handleData')
  const wrapper = mount(<Main />);
  wrapper.setState({ ...defaultState });
  const instance = wrapper.instance() as Main;

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Check correct left panel is showing", () => {
    expect(instance.state.isPlanning).toEqual(false);
    expect(wrapper.find("RunLayout")).toBeDefined();
  });

  it("ComponentDidMount", async () => {
    const spy = jest.spyOn(Main.prototype, "componentDidMount");
    const wrapper = mount(<Main />);
    await expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it("API test", async () => {
    const response = await (
      await PicarroAPI.getRequest(`http://${apiLoc}/uistatus`)
    ).json();
    expect(response).toEqual(uistatus.uistatus);
  });

  it("handleData", () => {
    instance.handleData(JSON.stringify(uistatus));
    instance.handleData(JSON.stringify(plan));
    instance.handleData(JSON.stringify(modal_info));
    instance.handleData(JSON.stringify({}));
    instance.setState({ initialized: false });
    instance.handleData(JSON.stringify({}));
  });

  it("WS", () => {
    instance.ws_sender(uistatus);
    expect(instance.ws.send).toBeTruthy();
  });

  it("Test Toast", () => {
    notifyError(
      <div>
        <h6>
          <b>Web Socket Disconnected!</b>
        </h6>
      </div>
    );
    const toast = wrapper.find("ToastContainer");
    expect(toast).toMatchSnapshot();
    notifySuccess(
      <div>
        <h6>
          <b>Web Socket Connected!</b>
        </h6>
      </div>
    );
    const toastSuccess = wrapper.find("ToastContainer");
    expect(toastSuccess).toMatchSnapshot();
  });

  it("Test WS", async ()=>{
    const data = '{"uistatus": {"timer": 5 }}'
    instance.ws = client;
    instance.attachWSMethods(instance.ws);
    await server.connected;
    console.log(data)
    console.log(JSON.parse(data))
    server.send(data);
    expect(handleDataSpy).toHaveBeenCalled();
    server.close();
    instance.ws.close();
    // expect(client).toReceiveMessage();
  })

  // server.close;
  // client.close;
});
