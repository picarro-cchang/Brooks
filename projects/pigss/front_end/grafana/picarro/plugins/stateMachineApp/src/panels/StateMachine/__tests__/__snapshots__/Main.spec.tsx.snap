// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Snapshot 1`] = `
<Main>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row justify-content-md-center"
      >
        <div
          className="col-sm-3"
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <CommandPanel
            plan={
              Object {
                "available_ports": Object {
                  "1": 170,
                  "2": 0,
                  "3": 170,
                  "4": 170,
                },
                "bank_names": Object {
                  "1": Object {
                    "channels": Object {
                      "1": "Ch. 1",
                      "2": "Ch. 2",
                      "3": "Ch. 3",
                      "4": "Ch. 4",
                      "5": "Ch. 5",
                      "6": "Ch. 6",
                      "7": "Ch. 7",
                      "8": "Ch. 8",
                    },
                    "name": "Bank 1",
                  },
                  "2": Object {
                    "channels": Object {
                      "1": "Ch. 1",
                      "2": "Ch. 2",
                      "3": "Ch. 3",
                      "4": "Ch. 4",
                      "5": "Ch. 5",
                      "6": "Ch. 6",
                      "7": "Ch. 7",
                      "8": "Ch. 8",
                    },
                    "name": "Bank 2",
                  },
                  "3": Object {
                    "channels": Object {
                      "1": "Ch. 1",
                      "2": "Ch. 2",
                      "3": "Ch. 3",
                      "4": "Ch. 4",
                      "5": "Ch. 5",
                      "6": "Ch. 6",
                      "7": "Ch. 7",
                      "8": "Ch. 8",
                    },
                    "name": "Bank 3",
                  },
                  "4": Object {
                    "channels": Object {
                      "1": "Ch. 1",
                      "2": "Ch. 2",
                      "3": "Ch. 3",
                      "4": "Ch. 4",
                      "5": "Ch. 5",
                      "6": "Ch. 6",
                      "7": "Ch. 7",
                      "8": "Ch. 8",
                    },
                    "name": "Bank 4",
                  },
                },
                "current_step": 11,
                "focus": Object {
                  "column": 1,
                  "row": 12,
                },
                "last_step": 11,
                "looping": false,
                "max_steps": 32,
                "num_plan_files": 2,
                "panel_to_show": 0,
                "plan_filename": "__default__",
                "plan_files": Object {
                  "1": "__default__",
                  "2": "tr",
                },
                "steps": Object {
                  "1": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 2,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "10": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 8,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "11": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 128,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "2": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 32,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "3": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 8,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "4": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 128,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "5": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 32,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "6": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 8,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "7": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 128,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "8": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 2,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "9": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 32,
                        "clean": 0,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                },
              }
            }
            uistatus={Object {}}
            ws_sender={[Function]}
          >
            <div
              className="panel-command"
            >
              <div
                style={
                  Object {
                    "marginTop": 20,
                    "width": "100%",
                  }
                }
              >
                <div
                  className="grid-command"
                  style={
                    Object {
                      "display": "grid",
                      "gridGap": 5,
                      "gridTemplateColumns": "1fr 1fr 1fr",
                      "padding": 20,
                    }
                  }
                >
                  <button
                    className="btn btn-large btn-command btn-1-3 "
                    disabled={true}
                    id="standby"
                    onClick={[Function]}
                    value="standby"
                  >
                    Standby
                  </button>
                  <button
                    className="btn btn-large btn-command btn-1-3 "
                    disabled={true}
                    id="identify"
                    onClick={[Function]}
                    value="identify"
                  >
                    Identify Available Channels
                  </button>
                  <button
                    className="btn btn-large btn-command btn-1-3 "
                    disabled={true}
                    id="edit-plan"
                    onClick={[Function]}
                    value="plan"
                  >
                    Edit Plan
                  </button>
                  <button
                    className="btn btn-large btn-command btn-run "
                    disabled={true}
                    id="run-channel"
                    onClick={[Function]}
                    value="run"
                  >
                    Run Channel
                  </button>
                  <button
                    className="btn btn-large btn-command btn-run "
                    disabled={true}
                    id="run-plan"
                    onClick={[Function]}
                    value="plan_run"
                  >
                    Run Plan
                  </button>
                  <button
                    className="btn btn-large btn-command btn-run "
                    disabled={true}
                    id="loop-plan"
                    onClick={[Function]}
                    value="plan_loop"
                  >
                    Loop Plan
                  </button>
                  <button
                    className="btn btn-large btn-command btn-1-3 "
                    disabled={true}
                    id="reference"
                    onClick={[Function]}
                    value="reference"
                  >
                    Reference
                  </button>
                  <button
                    className="btn btn-large btn-edit btn-danger"
                    disabled={true}
                    id="edit-labels"
                    onClick={[Function]}
                    value="edit"
                  >
                    Edit Labels
                  </button>
                </div>
              </div>
            </div>
          </CommandPanel>
        </div>
        <div
          className="col-sm-9"
          style={
            Object {
              "display": "grid",
              "gridTemplateColumns": "1fr",
            }
          }
        >
          <div
            style={
              Object {
                "gridColumnStart": "1",
                "gridRowStart": "1",
                "padding": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <Modal
      animationDuration={500}
      blockScroll={true}
      center={true}
      classNames={Object {}}
      classes={
        Object {
          "closeButton": "styles_closeButton__20ID4",
          "closeIcon": "styles_closeIcon__1QwbI",
          "modal": "styles_modal__gNwvD",
          "modalCenter": "styles_modalCenter__L9F2w",
          "overlay": "styles_overlay__CLSq-",
          "transitionEnter": "styles_transitionEnter__3j_-a",
          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
          "transitionExit": "styles_transitionExit__1KmEf",
          "transitionExitActive": "styles_transitionExitActive__1nQXw",
        }
      }
      closeIconSize={28}
      closeIconSvgPath={
        <path
          d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
        />
      }
      closeOnEsc={true}
      closeOnOverlayClick={true}
      focusTrapOptions={Object {}}
      focusTrapped={true}
      onClose={[Function]}
      open={false}
      role="dialog"
      showCloseIcon={true}
      styles={
        Object {
          "overlay": Object {
            "color": "black",
          },
        }
      }
    />
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    >
      <div
        className="Toastify"
      />
    </ToastContainer>
  </div>
</Main>
`;

exports[`<Main /> Test Toast 1`] = `
<ToastContainer
  autoClose={5000}
  bodyClassName={null}
  className={null}
  closeButton={
    <CloseButton
      ariaLabel="close"
    />
  }
  closeOnClick={true}
  draggable={true}
  draggablePercent={80}
  hideProgressBar={false}
  newestOnTop={false}
  pauseOnFocusLoss={true}
  pauseOnHover={true}
  position="top-right"
  progressClassName={null}
  progressStyle={null}
  role="alert"
  rtl={false}
  style={null}
  toastClassName={null}
  transition={[Function]}
>
  <div
    className="Toastify"
  />
</ToastContainer>
`;

exports[`<Main /> Test Toast 2`] = `
<ToastContainer
  autoClose={5000}
  bodyClassName={null}
  className={null}
  closeButton={
    <CloseButton
      ariaLabel="close"
    />
  }
  closeOnClick={true}
  draggable={true}
  draggablePercent={80}
  hideProgressBar={false}
  newestOnTop={false}
  pauseOnFocusLoss={true}
  pauseOnHover={true}
  position="top-right"
  progressClassName={null}
  progressStyle={null}
  role="alert"
  rtl={false}
  style={null}
  toastClassName={null}
  transition={[Function]}
>
  <div
    className="Toastify"
  />
</ToastContainer>
`;

exports[`<Main w/ PlanPanel /> Snapshot 1`] = `
<Main>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row justify-content-md-center"
      >
        <div
          className="col-sm-3"
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <PlanPanel
            isChanged={false}
            plan={
              Object {
                "bank_names": Object {
                  "1": Object {
                    "channels": Object {
                      "1": "Channel 1",
                      "2": "Channel 2",
                      "3": "Channel 3",
                      "4": "Channel 4",
                      "5": "Channel 5",
                      "6": "Channel 6",
                      "7": "Channel 7",
                      "8": "Channel 8",
                    },
                    "name": "Bank 1",
                  },
                  "2": Object {
                    "channels": Object {
                      "1": "Channel 1",
                      "2": "Channel 2",
                      "3": "Channel 3",
                      "4": "Channel 4",
                      "5": "Channel 5",
                      "6": "Channel 6",
                      "7": "Channel 7",
                      "8": "Channel 8",
                    },
                    "name": "Bank 2",
                  },
                  "3": Object {
                    "channels": Object {
                      "1": "Channel",
                      "2": "Channel",
                      "3": "Channel",
                      "4": "Channel",
                      "5": "Channel",
                      "6": "Channel",
                      "7": "Channel",
                      "8": "Channel",
                    },
                    "name": "Bank 3",
                  },
                  "4": Object {
                    "channels": Object {
                      "1": "Channel",
                      "2": "Channel",
                      "3": "Channel",
                      "4": "Channel",
                      "5": "Channel",
                      "6": "Channel",
                      "7": "Channel",
                      "8": "Channel",
                    },
                    "name": "Bank 4",
                  },
                },
                "current_step": 1,
                "focus": Object {
                  "column": 2,
                  "row": 1,
                },
                "last_step": 4,
                "max_steps": 32,
                "num_plan_files": 1,
                "panel_to_show": 1,
                "plan_filename": "test.pln",
                "plan_files": Object {
                  "1": "test.pln",
                },
                "steps": Object {
                  "1": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 1,
                    "reference": 1,
                  },
                  "2": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 0,
                        "clean": 10,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 10,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 10,
                      },
                    },
                    "duration": 20,
                    "reference": 0,
                  },
                  "3": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 8,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 60,
                    "reference": 0,
                  },
                  "4": Object {
                    "banks": Object {
                      "1": Object {
                        "chan_mask": 128,
                        "clean": 0,
                      },
                      "3": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                      "4": Object {
                        "chan_mask": 0,
                        "clean": 0,
                      },
                    },
                    "duration": 60,
                    "reference": 0,
                  },
                },
              }
            }
            setFocus={[Function]}
            uistatus={Object {}}
            updateFileName={[Function]}
            ws_sender={[Function]}
          >
            <div>
              <div
                className="panel-plan"
              >
                <h2
                  className="panel-plan-text"
                >
                  Schedule
                </h2>
                <span
                  className="cancel panel-plan-text"
                  id="cancel-x"
                  onClick={[Function]}
                />
                <h6
                  className="panel-plan-text"
                >
                  Please click on available channels to set up a schedule, then click on the radio button to select starting position.
                </h6>
                <div>
                  <h6
                    className="panel-plan-text"
                  >
                    Currently viewing File:
                     
                    <span
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      test.pln
                    </span>
                  </h6>
                </div>
                <div
                  className="panel-plan-inner"
                >
                  <form>
                    <ReactList
                      axis="y"
                      itemRenderer={[Function]}
                      itemsRenderer={[Function]}
                      length={32}
                      minSize={1}
                      pageSize={10}
                      scrollParentGetter={[Function]}
                      scrollParentViewportSizeGetter={[Function]}
                      threshold={100}
                      type="uniform"
                      useStaticSize={false}
                      useTranslate3d={false}
                    >
                      <div
                        style={
                          Object {
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "WebkitTransform": "translate(0px, 0px)",
                              "msTransform": "translate(0px, 0px)",
                              "transform": "translate(0px, 0px)",
                            }
                          }
                        >
                          <div>
                            <div
                              key="0"
                            >
                              <div
                                className="gf-form-inline"
                                key="1"
                              >
                                <label
                                  className="col-sm-1 panel-plan-text"
                                  style={
                                    Object {
                                      "fontSize": "17px",
                                      "marginTop": "5px",
                                    }
                                  }
                                >
                                  1. 
                                </label>
                                <div
                                  className="col-sm-6 col-bank-name"
                                >
                                  <input
                                    className="form-control plan-input panel-plan-text"
                                    id="plan-port-1"
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    placeholder="Select port"
                                    style={
                                      Object {
                                        "float": "left",
                                        "marginLeft": "2px",
                                        "maxWidth": "90%",
                                      }
                                    }
                                    type="text"
                                    value="Reference"
                                  />
                                </div>
                                <div
                                  className="col-sm-3"
                                  style={
                                    Object {
                                      "paddingLeft": "0px",
                                    }
                                  }
                                >
                                  <input
                                    className="form-control input-small plan-input panel-plan-text"
                                    id="plan-duration-1"
                                    maxLength={8}
                                    minLength={1}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    placeholder="Duration"
                                    style={
                                      Object {
                                        "maxWidth": "100%",
                                      }
                                    }
                                    type="text"
                                    value="1"
                                  />
                                </div>
                                <label
                                  className="panel-plan-text"
                                  style={
                                    Object {
                                      "marginLeft": "-15px",
                                      "paddingRight": "5px",
                                    }
                                  }
                                >
                                  s
                                </label>
                                <label
                                  className="col-sm-1 radio-btn"
                                >
                                  <input
                                    checked={true}
                                    id="plan-row-1"
                                    onChange={[Function]}
                                    style={
                                      Object {
                                        "maxWidth": "100%",
                                      }
                                    }
                                    type="radio"
                                  />
                                  <span
                                    className="checkmark"
                                  />
                                </label>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </ReactList>
                  </form>
                </div>
                <div
                  className="container"
                >
                  <div
                    className="row text-center btn-row-1"
                  >
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-group"
                        disabled={false}
                        id="insert-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        Insert
                      </button>
                    </div>
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-light btn-group"
                        id="save-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-light btn-group"
                        id="load-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        Load
                      </button>
                    </div>
                  </div>
                  <div
                    className="row btn-row-2"
                  >
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-cancel btn-group"
                        disabled={false}
                        id="delete-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete
                      </button>
                    </div>
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-cancel btn-group"
                        disabled={false}
                        id="clear-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        Clear
                      </button>
                    </div>
                    <div
                      className="col-sm-3"
                    >
                      <button
                        className="btn btn-block btn-green btn-group"
                        id="ok-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        OK
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </PlanPanel>
        </div>
        <div
          className="col-sm-9"
          style={
            Object {
              "display": "grid",
              "gridTemplateColumns": "1fr",
            }
          }
        >
          <div
            style={
              Object {
                "gridColumnStart": "1",
                "gridRowStart": "1",
                "padding": "10px",
              }
            }
          />
          <div
            className="ref-div"
          >
            <button
              className="btn btn-large ref-btn"
              id="reference"
              onClick={[Function]}
              style={
                Object {
                  "color": "black",
                }
              }
              type="button"
            >
              Reference
            </button>
          </div>
        </div>
      </div>
    </div>
    <Modal
      animationDuration={500}
      blockScroll={true}
      center={true}
      classNames={Object {}}
      classes={
        Object {
          "closeButton": "styles_closeButton__20ID4",
          "closeIcon": "styles_closeIcon__1QwbI",
          "modal": "styles_modal__gNwvD",
          "modalCenter": "styles_modalCenter__L9F2w",
          "overlay": "styles_overlay__CLSq-",
          "transitionEnter": "styles_transitionEnter__3j_-a",
          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
          "transitionExit": "styles_transitionExit__1KmEf",
          "transitionExitActive": "styles_transitionExitActive__1nQXw",
        }
      }
      closeIconSize={28}
      closeIconSvgPath={
        <path
          d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
        />
      }
      closeOnEsc={true}
      closeOnOverlayClick={true}
      focusTrapOptions={Object {}}
      focusTrapped={true}
      onClose={[Function]}
      open={false}
      role="dialog"
      showCloseIcon={true}
      styles={
        Object {
          "overlay": Object {
            "color": "black",
          },
        }
      }
    />
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    >
      <div
        className="Toastify"
      />
    </ToastContainer>
  </div>
</Main>
`;
